driver/src/main/java/org/neo4j/driver/internal/SecuritySettings.java

```diff
@@ -0,0 +1,199 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.driver.internal;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+import org.neo4j.driver.Config;
+import org.neo4j.driver.exceptions.ClientException;
+import org.neo4j.driver.internal.security.SecurityPlan;
+import org.neo4j.driver.internal.security.SecurityPlanImpl;
+
+import static org.neo4j.driver.internal.Scheme.isHighTrustScheme;
+import static org.neo4j.driver.internal.Scheme.isLowTrustScheme;
+import static org.neo4j.driver.internal.Scheme.isSecurityScheme;
+import static org.neo4j.driver.internal.security.SecurityPlanImpl.insecure;
+
+public class SecuritySettings
+{
+    private static final boolean DEFAULT_ENCRYPTED = false;
+    private static final Config.TrustStrategy DEFAULT_TRUST_STRATEGY = Config.TrustStrategy.trustSystemCertificates();
+    private static final SecuritySettings DEFAULT = new SecuritySettings( DEFAULT_ENCRYPTED, DEFAULT_TRUST_STRATEGY );
+    private final boolean encrypted;
+    private final Config.TrustStrategy trustStrategy;
+
+    public SecuritySettings( boolean encrypted, Config.TrustStrategy trustStrategy )
+    {
+        this.encrypted = encrypted;
+        this.trustStrategy = trustStrategy == null ? DEFAULT_TRUST_STRATEGY : trustStrategy;
+    }
+
+    public boolean encrypted()
+    {
+        return encrypted;
+    }
+
+    public Config.TrustStrategy trustStrategy()
+    {
+        return trustStrategy;
+    }
+
+    private boolean isCustomized()
+    {
+        return this != DEFAULT;
+    }
+
+    public SecurityPlan createSecurityPlan( String uriScheme )
+    {
+        try
+        {
+            if ( isSecurityScheme( uriScheme ) )
+            {
+                assertSecuritySettingsNotUserConfigured( uriScheme );
+                return createSecurityPlanFromScheme( uriScheme );
+            }
+            else
+            {
+                return createSecurityPlanImpl( encrypted, trustStrategy );
+            }
+        }
+        catch ( GeneralSecurityException | IOException ex )
+        {
+            throw new ClientException( "Unable to establish SSL parameters", ex );
+        }
+    }
+
+    private void assertSecuritySettingsNotUserConfigured( String uriScheme )
+    {
+        if ( isCustomized() )
+        {
+            throw new ClientException( String.format( "Scheme %s is not configurable with manual encryption and trust settings", uriScheme ) );
+        }
+    }
+
+    private SecurityPlan createSecurityPlanFromScheme( String scheme ) throws GeneralSecurityException, IOException
```

Comment:
Can we simplify this method as:
```
if ( isHighTrustScheme(scheme) )
{ 
    return SecurityPlanImpl.forSystemCASignedCertificates( trustStrategy.isHostnameVerificationEnabled() ); 
}
else 
{
    return SecurityPlanImpl.forAllCertificates( trustStrategy.isHostnameVerificationEnabled() );
}
```

We've already checked `isSecurityScheme` and `assertSecuritySettingsNotUserConfigured`.

Reply:
of course! I will update this now